File: 448FindAllNumDisapeared.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        n = len(nums)
        dp = [True] * (n+1)
        for v in nums:
            dp[v] = False
        return [i for i in range(1, n+1) if dp[i]]


a = Solution()
print(a.findDisappearedNumbers([4, 3, 2, 7, 8, 2, 3, 1]
                               ))

File: 448FindAllNumDisapeared.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;
vector<int> findDisappearedNumbers(vector<int> &nums)
{
    vector<int> dp(nums.size() + 1);
    for (int i = 0; i < nums.size(); i++) {
        dp[nums[i]] = 1;
    }
    vector<int> res;
    for (int i = 1; i < dp.size(); i++) {
        if (dp[i] == 0) {
            res.push_back(i);
        }
    }
    return res;
}

int main()
{
    vector<int> a{2, 4, 5};
    findDisappearedNumbers(a);

    return 0;
}